.travis.yml

```diff
@@ -10,23 +10,18 @@ python:
 # Enable 3.7 without globally enabling sudo and dist: xenial for other build jobs
 # see this issue for updates https://github.com/travis-ci/travis-ci/issues/9815
 matrix:
-  allow_failures:
-    - python: 3.7
-      dist: xenial
-      sudo: true
   include:
     - python: 3.7
       dist: xenial
       sudo: true
 
-install:
+before_install:
   - pip install -r requirements.txt
   - pip install -r test-requirements.txt
-  - pip install pytest coverage codecov
-  - pip install -e .
-  # Need to clean up the git repository or pytest gets confused by the extra
-  # cruft that python setup.py install leaves behind
-#  - git clean -xfd
+
+install:
+  - python setup.py sdist && version=$(python setup.py --version) && pushd dist && pip install depfinder-${version}.tar.gz && popd
```

Comment:
what's the advantage of doing this all on one line?

Reply:
This creates the tarball and install from it, that way we can catch errors that may happen due to a bad `MANIFEST.in` or other packaging errors. This is not too important and I can remove it if you don't like it.

Reply:
no it doesn't bother me at all. I was mostly just wondering the advantage of all the `&&` to string a bunch of bash together.

Reply:
Yeah. IMO `pip` is broken for not checking those when installing from source and not a tarball. I believe this is "fixed" with the `pyproject.toml` b/c you can have `MANIFEST.in`, dependencies, etc all in a single file that is interpreted by `pip`. Not sure if that is working as expected though b/c I never used it.